bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TupleValueImpl.java

```diff
@@ -594,7 +592,6 @@ public void freezeDirect() {
                 ((RefValue) value).freezeDirect();
             }
         }
-        this.typedesc = createSingletonTypedesc(this);
```

Comment:
Can set this `null` if removing the line is not okay. If this is not removed may need to change in subsequent places also.